components/admin_console/system_user_detail/system_user_detail.jsx

```diff
@@ -0,0 +1,374 @@
+                        id='deactivate_member_modal.sso_warning'
+                        defaultMessage='You must also deactivate this user in the SSO provider or they will be reactivated on next login or sync.'
+                    />
+                </strong>
+            );
+        }
+
+        const message = (
+            <div>
+                <FormattedMessage
+                    id='deactivate_member_modal.desc'
+                    defaultMessage='This action deactivates {username}. They will be logged out and not have access to any teams or channels on this system. Are you sure you want to deactivate {username}?'
+                    values={{
+                        username: user.username,
+                    }}
+                />
+                {warning}
+            </div>
+        );
+
+        const confirmButtonClass = 'btn btn-danger';
+        const deactivateMemberButton = (
+            <FormattedMessage
+                id='deactivate_member_modal.deactivate'
+                defaultMessage='Deactivate'
+            />
+        );
+
+        return (
+            <ConfirmModal
+                show={this.state.showDeactivateMemberModal}
+                title={title}
+                message={message}
+                confirmButtonClass={confirmButtonClass}
+                confirmButtonText={deactivateMemberButton}
+                onConfirm={this.handleDeactivateMember}
+                onCancel={this.handleDeactivateCancel}
+            />
+        );
+    }
+
+    renderActivateDeactivate = () => {
+        if (this.props.user.delete_at > 0) {
+            return (
+                <AdminButtonOutline
+                    onClick={this.handleMakeActive}
+                    className='admin-btn-default'
+                >
+                    {Utils.localizeMessage('admin.user_item.makeActive', 'Activate')}
+                </AdminButtonOutline>
+            );
+        }
+        return (
+            <AdminButtonOutline
+                onClick={this.handleShowDeactivateMemberModal}
+                className='admin-btn-default'
+            >
+                {Utils.localizeMessage('admin.user_item.makeInactive', 'Deactivate')}
+            </AdminButtonOutline>
+        );
+    }
+
+    renderRemoveMFA = () => {
+        if (this.props.user.mfa_active) {
+            return (
+                <AdminButtonOutline
+                    onClick={this.handleResetMfa}
+                    className='admin-btn-default'
+                >
+                    {'Remove MFA'}
+                </AdminButtonOutline>
+            );
+        }
+        return null;
+    }
+
+    render() {
+        const {user} = this.props;
+        let deactivateMemberModal;
+        let currentRoles = (
+            <FormattedMessage
+                id='admin.user_item.member'
+                defaultMessage='Member'
+            />
+        );
+
+        // TODO: get the user object if navigated to page directly
```

Comment:
Can we convert it into a ticket and remove the TODO from the code?

Reply:
That's a good idea. I'll create a ticket.

Reply:
https://mattermost.atlassian.net/browse/MM-16824